services/shhext/chat/encryption.go

```diff
@@ -238,8 +239,8 @@ func (s *EncryptionService) DecryptPayload(myIdentityKey *ecdsa.PrivateKey, thei
 	}
 
 	// We should not be sending a signal if it's coming from us, as we receive our own messages
-	if msg == nil {
-		return nil, ErrSessionNotFound
+	if msg == nil && myIdentityKey.PublicKey != *theirIdentityKey {
```

Comment:
I feel it's more readable if we switch the operands in the comparison:

```suggestion
	if msg == nil && *theirIdentityKey != myIdentityKey.PublicKey {
```